addons/account/models/mail_message.py

```diff
@@ -5,21 +5,32 @@
 
 bypass_token = object()
 DOMAINS = {
-    'account.move': lambda operator, value: [('company_id.check_account_audit_trail', operator, value)],
-    'account.account': lambda operator, value: [('company_ids.check_account_audit_trail', operator, value)],
-    'account.tax': lambda operator, value: [('company_id.check_account_audit_trail', operator, value)],
-    'res.partner': lambda operator, value: [
-        '|', ('company_id', '=', False), ('company_id.check_account_audit_trail', operator, value),
-        '|', ('customer_rank', '>', 0), ('supplier_rank', '>', 0),
-    ],
-    'res.company': lambda operator, value: [('check_account_audit_trail', operator, value)],
+    'res.company':
+        lambda operator, value, extra_ids=None:
+            [('id', 'in', extra_ids)] if extra_ids is not None else [],
+    'account.move':
+        lambda operator, value, extra_ids=None:
+            [('company_id.restrictive_audit_trail', operator, value)],
+    'account.account':
+        lambda operator, value, extra_ids=None:
+            [('used', operator, value), ('company_ids.restrictive_audit_trail', operator, value)],
+    'account.tax':
+        lambda operator, value, extra_ids=None:
+            ([('id', 'in', extra_ids)] if extra_ids is not None else []),
+    'res.partner':
+        lambda operator, value, extra_ids=None:
+            ['|', ('customer_rank', '>', 0), ('supplier_rank', '>', 0)] + ([('id', 'in', extra_ids)] if extra_ids is not None else []),
```

Comment:
You don't need the customer/supplier ranks if you are already getting the extra ids from a query on `account.move.line`